src/PrestaShopBundle/Bridge/AdminController/LegacyControllerBridge.php

```diff
@@ -0,0 +1,306 @@
+     */
+    public function addJS($jsUri, $checkPath = true): void
+    {
+        if (!is_array($jsUri)) {
+            $jsUri = [$jsUri];
+        }
+
+        foreach ($jsUri as $jsFile) {
+            $jsFile = explode('?', $jsFile);
+            $version = '';
+            if (isset($jsFile[1]) && $jsFile[1]) {
+                $version = $jsFile[1];
+            }
+            $jsPath = $jsFile = $jsFile[0];
+            if ($checkPath) {
+                $jsPath = Media::getJSPath($jsFile);
+            }
+
+            if ($jsPath && !in_array($jsPath, $this->controllerConfiguration->jsFiles)) {
+                $this->controllerConfiguration->jsFiles[] = $jsPath . ($version ? '?' . $version : '');
+            }
+        }
+    }
+
+    /**
+     * {@intheritedoc}
+     */
+    public function addJqueryPlugin($name, $folder = null, $css = true): void
+    {
+        if (!is_array($name)) {
+            $name = [$name];
+        }
+
+        foreach ($name as $plugin) {
+            $pluginPath = Media::getJqueryPluginPath($plugin, $folder);
+
+            if (!empty($pluginPath['js'])) {
+                $this->addJS($pluginPath['js'], false);
+            }
+            if ($css && !empty($pluginPath['css'])) {
+                if (is_array($pluginPath['css'])) {
+                    $this->addCSS((string) key($pluginPath['css']), 'all', null, false);
+                } else {
+                    $this->addCSS($pluginPath['css'], 'all', null, false);
+                }
+            }
+        }
+    }
+
+    /**
+     * {@intheritedoc}
+     */
+    public function addJqueryUI($component, $theme = 'base', $checkDependencies = true): void
+    {
+        if (!is_array($component)) {
+            $component = [$component];
+        }
+
+        foreach ($component as $ui) {
+            $uiPath = Media::getJqueryUIPath($ui, $theme, $checkDependencies);
+            $this->addCSS($uiPath['css'], 'all');
+            $this->addJS($uiPath['js'], false);
+        }
+    }
+
+    /**
+     * This whole bridge is used as legacy $context->controller, but all properties are held in configuration,
+     * so we use a "magic" getter to allow legacy code retrieving properties from configuration as if it would be in legacy controller
+     *
+     * @param string $name
+     *
+     * @return mixed
+     */
+    public function __get(string $name)
+    {
+        if (property_exists($this, $name)) {
+            return $this->{$name};
+        }
+
+        return $this->controllerConfiguration->{$name};
+    }
+
+    /**
+     * This whole bridge is used as legacy $context->controller, but all properties are held in configuration,
+     * so we use a "magic" setter to allow legacy code setting properties in configuration as if it would be in legacy controller.
+     *
+     * @todo: for setter "property_exists" will probably raise issues, because legacy used to set previously not defined properties in controller too (so this wouldn't work)
```

Comment:
Maybe you can use the PropertyAccessor of Symfony, because it will look magic methods